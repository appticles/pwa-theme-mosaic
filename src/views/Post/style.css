.post.container {
  display: flex;
  flex-direction: column;
  margin: 0 !important;
}

.post.container .container {
  margin: 0 1em !important;
}
.post.container .container .ui.header {
  text-align: start;
  color: #000000;
  height: unset;
  position: static;
  font-size: 2.4rem;
  font-weight: 100;
}
.post.container .sub.header {
  height: 6%;
  color: #999;
  margin: -3% 0% 5% 7%;
}
.post.container .sub.header b {
  font-size: 15px;
  color: #000000;
}

.ui.justified.container {
  height: unset;
}

.post.container .container .label {
  height: unset;
  border-radius: 0;
  font-size: 1.01rem;
  font-weight: 100;
  padding: 10px 5px;
  margin: 15px 10px -10px 0;
}

.post .post-image {
  height: 60vh;
  background: center no-repeat;
  background-size: cover;
}

/* ICON */
.ui.container.post i.icon {
  color: #ffffff;
}

.ui.container.post i.icon.big.chevron.left {
  position: absolute;
  z-index: 1;
  top: 10px;
  left: 0;
}

.ui.container.post i.icon.circular.share {
  background-color: rgba(116, 116, 116, 0.59);
  font-size: 1.4em;
  margin: 0 10px 0 0;
  box-shadow: none;
}

.transition-container {
  display: flex;
  justify-content: flex-end;
}

/* FIX FOR EXTRA DIV FORM react-router-modal */
.ui.container.post + div {
  display: none;
}

/* MODAL */
.scrolling.dimmable.dimmed > .dimmer {
  overflow: hidden;
}
.dimmed.dimmable > .ui.animating.dimmer,
.dimmed.dimmable > .ui.visible.dimmer,
.ui.active.dimmer {
  opacity: 0.9;
}

.ui.page.modals.dimmer.transition.visible.active .actions {
  display: flex;
  justify-content: space-evenly;
  align-items: flex-end;
  height: 93%;
}
.ui.page.modals.dimmer.transition.visible.active .social {
  justify-content: space-evenly;
  display: flex;
}
