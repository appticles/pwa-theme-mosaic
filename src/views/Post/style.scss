@import "../../utilities/variables";

.post {
  &.container {
    display: flex;
    flex-direction: column;
    margin: 0 !important;
    text-align: left;

    .post-image {
      height: 60vh;
      background: center no-repeat;
      background-size: cover;
    }

    .container {
      margin: $base-spacing;
      padding-top: $base-spacing-sm;

      .ui.header {
        text-align: start;
      }

      .sub.header {
        margin-top: $base-spacing;
        margin-bottom: $base-spacing;
      }

      .label {
        height: unset;
        border-radius: 2px;
        padding: $base-spacing-xsm $base-spacing-xs;
        margin: $base-spacing-xs;
        background-color: $category-color;
        color: $category-text;
      }

      .post-content {
        padding: 0 $base-spacing-xsm;
        iframe {
          max-width: 100%;
          height: auto;
        }
        img {
          max-width: 100%;
          height: auto;
        }

        div {
          max-width: 100%;
        }
      }
    }
    /* ICON */
    i {
      &.icon.circular {
        color: $buttons-icon-color;
        background-color: $buttons-bg-color;
        box-shadow: none;

        &.chevron.left {
          margin-top: $base-spacing;
          margin-left: $base-spacing;
        }

        &.chevron.left.absolute {
          position: absolute;
          z-index: 1;
          top: $base-spacing;
          left: $base-spacing;
          margin-top: 0;
          margin-left: 0;
        }

        &.share {
          margin: $base-spacing-md;
        }
      }
    }
  }
  .transition-container {
    display: flex;
    justify-content: flex-end;
  }
}
