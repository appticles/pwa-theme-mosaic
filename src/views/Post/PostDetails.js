import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Helmet from 'react-helmet';
import { Container, Header, Icon, Modal, Transition } from 'semantic-ui-react';
import Moment from 'react-moment';
import PropTypes from 'prop-types';
import _ from 'lodash';
import config from 'react-global-configuration';
import { DFPSlotsProvider, AdSlot } from 'react-dfp';
import SocialMedia from './components/SocialMedia';
import { postPropType } from '../PostsCarousel/reducer';
import CategoryLabel from '../../components/CategoryLabel';
import './style.css';

class PostDetails extends Component {
  constructor(props) {
    super(props);

    this.handleOpen = this.handleOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);

    this.state = {
      modalOpen: false,
      visible: true,
    };
  }

  getPostContent() {
    const { post } = this.props;
    if (post.content.protected === false) {
      return post.content.rendered;
    }

    return (
      "<p>This post is password protected! Enter the password <a href='" +
      post.link +
      "?pwapp_theme_mode=desktop'>here</a>."
    );
  }

  handleOpen() {
    this.setState({ modalOpen: true, visible: !this.state.visible });
  }

  handleClose() {
    this.setState({ modalOpen: false, visible: !this.state.visible });
  }

  render() {
    const { post } = this.props;
    const { author } = post._embedded;
    const categoriesList = post._embedded['wp:term'];
    const featuredMedia = post._embedded['wp:featuredmedia'];

    // adsense params
    const googleAds = config.get('googleAds');

    // set path routes
    let goBack = '/';
    let path = '';

    if (!_.isNil(this.props.category.categorySlug) && !_.isNil(this.props.category.categoryId)) {
      path = '/category/' + this.props.category.categorySlug + '/' + this.props.category.categoryId;
      goBack = path;
    }

    path = path + '/post/' + post.slug + '/' + post.id + '/comments/' + post.comment_status;

    return (
      <Container className="post">
        <Helmet>
          <link rel="canonical" href={post.link} />
        </Helmet>
        <Link to={goBack}>
          <Icon circular size="large" name="chevron left" className={featuredMedia ? 'absolute' : ''} />
        </Link>
        {_.isArray(featuredMedia) && !_.isNil(featuredMedia[0]) ? (
          <div
            className="post-image"
            style={{ backgroundImage: `url(${featuredMedia[0].source_url})` }}
          />
        ) : null}
        <Container textAlign="justified">
          <Header>
            <h1 dangerouslySetInnerHTML={{ __html: post.title.rendered }} />
          </Header>
          {categoriesList[0].map(category => (
            <CategoryLabel
              link={`/category/${category.slug}/${category.id}`}
              name={category.name}
              key={category.name}
            />
          ))}
          <Header.Subheader>
            &nbsp;{this.props.texts.TEXTS && this.props.texts.TEXTS.BY_AUTHOR}&nbsp;<b>
              {author.map(item => item.name).join(', ')}
            </b>,&nbsp;<Moment format="MMMM DD, YYYY">{post.date}</Moment>
          </Header.Subheader>
          <div
            className="post-content"
            dangerouslySetInnerHTML={{ __html: this.getPostContent() }}
          />
          {!_.isNil(googleAds) && !_.isNil(googleAds.phone) ? (
            <DFPSlotsProvider
              dfpNetworkId={googleAds.phone.networkCode}
              sizeMapping={[{ viewport: [900, 768], sizes: googleAds.phone.sizes }]}
            >
              <div className="mobile-ads">
                <AdSlot sizes={googleAds.phone.sizes} adUnit={googleAds.phone.adUnitCode} />
              </div>
            </DFPSlotsProvider>
          ) : null}
        </Container>
        {post.content.protected === false ? (
          <Modal
            className="share"
            open={this.state.modalOpen}
            onClick={this.handleClose}
            trigger={
              <Transition.Group animation="fade up" duration="500" className="transition-container">
                {this.state.visible && (
                  <Icon circular name="share alternate" size="large" onClick={this.handleOpen} />
                )}
              </Transition.Group>
            }
            basic
          >
            <Modal.Actions>
              <Link to={path}>
                <Icon circular name="comment" size="large" />
              </Link>
              <SocialMedia title={post.title.rendered} link={post.link} />
            </Modal.Actions>
          </Modal>
        ) : null}
      </Container>
    );
  }
}

PostDetails.propTypes = {
  post: postPropType.isRequired,
  category: PropTypes.shape({
    categorySlug: PropTypes.string,
    categoryId: PropTypes.string,
  }).isRequired,
  texts: PropTypes.shape({
    TEXTS: PropTypes.shape({
      BY_AUTHOR: PropTypes.string,
    }),
  }),
};

PostDetails.defaultProps = {
  texts: {
    TEXTS: {
      BY_AUTHOR: 'by',
    },
  },
};

export default PostDetails;
