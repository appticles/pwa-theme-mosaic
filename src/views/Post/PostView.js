import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Container, Image, Header, Label, Icon, Modal, Transition } from 'semantic-ui-react';
import Moment from 'react-moment';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { postPropType } from '../PostsCarousel/reducer';
import './style.css';

export default class PostView extends Component {
  constructor(props) {
    super(props);
    this.getImage = this.getImage.bind(this);
    this.handleOpen = this.handleOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.state = {
      modalOpen: false,
      visible: true,
    };
  }

  getImage(sourceImg) {
    let imageSource;
    if (sourceImg) {
      imageSource = sourceImg[0].source_url;
      return imageSource;
    }
    imageSource = 'https://placeholdit.co//i/555x650';
    return imageSource;
  }

  handleOpen() {
    this.setState({ modalOpen: true, visible: !this.state.visible });
  }

  handleClose() {
    this.setState({ modalOpen: false, visible: !this.state.visible });
  }

  render() {
    const { post } = this.props;
    const { author } = post._embedded;
    const categoriesList = post._embedded['wp:term'];
    const featuredMedia = post._embedded['wp:featuredmedia'];
    // set path routes
    let goBack = {};
    let path = {};
    if (_.isNil(this.props.category.categorySlug)) {
      goBack = '/';
      path = '/post/' + post.slug + '/' + post.id + '/comments';
    } else {
      goBack = '/category/' + this.props.category.categorySlug + '/' + this.props.category.categoryId;
      path =
        '/category/' + this.props.category.categorySlug + '/' + this.props.category.categoryId + '/post/' + post.slug + '/' + post.id + '/comments';
    }

    return (
      <Container className="post">
        <Link to={goBack}>
          <Icon size="big" name="chevron left" />
        </Link>
        <Image src={this.getImage(featuredMedia)} />
        <Container textAlign="justified">
          {categoriesList[0].map(category => (
            <Label color="teal" key={category.name}>
              {category.name}
            </Label>
          ))}
          <Header>
            <div dangerouslySetInnerHTML={{ __html: post.title.rendered }} />
          </Header>
          <Header.Subheader>
            &nbsp;by&nbsp;<b>{author[0].name}</b>,&nbsp;<Moment format="MMMM DD, YYYY">{post.date}</Moment>
          </Header.Subheader>
          <div dangerouslySetInnerHTML={{ __html: post.content.rendered }} />
        </Container>
        <Modal
          open={this.state.modalOpen}
          onClick={this.handleClose}
          trigger={
            <Transition.Group animation="fade up" duration="500" className="transition-container">
              {this.state.visible && <Icon circular name="share alternate" size="large" onClick={this.handleOpen} />}
            </Transition.Group>
          }
          basic
        >
          <Modal.Actions>
            <Link to={path}>
              <Icon name="comment" size="large" circular inverted color="grey" />
            </Link>
            <a href={'https://m.facebook.com/sharer.php?u=' + post.link}>
              <Icon name="facebook f" size="large" circular inverted color="blue" />
            </a>
            <a href={'https://twitter.com/intent/tweet?text=' + encodeURIComponent(post.title.rendered) + ' ' + post.link}>
              <Icon name="twitter" size="large" circular inverted color="teal" />
            </a>
            <a href={'https://plus.google.com/share?url=' + post.link}>
              <Icon name="google plus" size="large" circular inverted color="red" />
            </a>
          </Modal.Actions>
        </Modal>
      </Container>
    );
  }
}
PostView.propTypes = {
  post: postPropType.isRequired,
  category: PropTypes.shape({
    categorySlug: PropTypes.string.isRequired,
    categoryId: PropTypes.string.isRequired,
  }).isRequired,
};
