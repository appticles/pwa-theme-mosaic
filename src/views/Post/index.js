import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import _ from 'lodash';

import { fetchPosts } from '../PostsCarousel/action';
import { getPosts, postPropType } from '../PostsCarousel/reducer';

import PostDetails from './PostView';

class Post extends Component {
  componentWillMount() {
    this.readPost(this.props.match.params.postId);
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.match.params.postId !== nextProps.match.params.postId) {
      this.readPost(nextProps.match.params.postId);
    }
  }

  readPost(postId) {
    const { dispatch } = this.props;
    dispatch(fetchPosts({ id: postId }));
  }

  render() {
    const post = this.props.posts.find(obj => obj.id === Number(this.props.match.params.postId));
    if (_.isNil(post)) {
      return <p>Post does not exist</p>;
    }

    return <PostDetails post={post} />;
  }
}

Post.propTypes = {
  dispatch: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      postId: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  posts: PropTypes.arrayOf(postPropType).isRequired,
};

const mapStateToProps = state => ({
  posts: getPosts(state.posts),
});

function mapDispatchToProps(dispatch) {
  return Object.assign({ dispatch }, bindActionCreators({ fetchPosts }, dispatch));
}

export default connect(mapStateToProps, mapDispatchToProps)(Post);
