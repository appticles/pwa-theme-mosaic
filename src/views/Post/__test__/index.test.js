import React from 'react';
import renderer from 'react-test-renderer';

import mockStore from '../../../__mocks__/__mocks__redux-mock-store';
import Post from '../index';

jest.mock('../PostView', () => 'Post view test');

describe('Post screen', () => {
  it('should render correctly that a post does not exist', () => {
    const store = mockStore({
      posts: {
        items: [],
        isFetching: 1,
      },
    });
    const match = {
      params: { postId: 'some-post-id' },
    };
    expect(renderer.create(<Post store={store} fetchPosts={jest.fn()} match={match} />)).toMatchSnapshot();
  });
  it('should render correctly an existing post', () => {
    const store = mockStore({
      posts: {
        items: [
          {
            id: 1,
            title: {
              rendered: 'Some post title',
            },
            author: 54,
            categories: [1, 4, 2],
            slug: 'some-post-slug',
            content: {
              rendered: 'Some post content to be rendered',
              protected: true,
            },
            excerpt: {
              rendered: 'Some post excerpt to be rendered',
              protected: false,
            },
            date: 'some mock date',
          },
        ],
        isFetching: 1,
      },
    });
    const match = {
      params: { postId: '1' },
    };
    expect(renderer.create(<Post store={store} fetchPosts={jest.fn()} match={match} />)).toMatchSnapshot();
  });
});
