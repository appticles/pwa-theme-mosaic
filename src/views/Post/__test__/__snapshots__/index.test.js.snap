// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post screen should render correctly an existing post 1`] = `
<p>
  Post does not exist
</p>
`;

exports[`Post screen should render correctly that a post does not exist 1`] = `
<p>
  Post does not exist
</p>
`;
